docs/generator.go

```diff
@@ -30,15 +31,15 @@ func main() {
 		}
 		file.Close()
 	}
-	f, err := os.OpenFile(tmpl, os.O_APPEND, 0666)
+	f, err := os.OpenFile(tmpl, os.O_APPEND|os.O_WRONLY, 0666)
```

Comment:
So do we want to have os.O_CREATE to also create the file? And do we really want it in append mode?

Reply:
Yes the .tmpl file does not exist beforehand and is also deleted in the last,so according to me yes it needs to be created L-28 each time we run the code.
Append because it will loop multiple times #L36-78 and to ensure the pointer does not reach the start in each itereation.

Reply:
Using os.O_CREATE should be enough to remove the manual file creation.

Ok makes sense to use append then, when we remove the files before.

Reply:
Yes so I assume no further thing is needed?

Reply:
We could try to remove the manual file creation and use os.O_CREATE.

Reply:
ack